### DISCUSSION TITLE
ID arg input conversion to Int before being passed to resolver

### DISCUSSION THREAD

**Discussion by solokhind** [hayes/pothos#1313] [Q&A] [answered]
Hi, I'm encountering an issue with ID scalars that’s adding unnecessary complexity to my resolvers.

In my database, primary keys are mostly integers with a few strings. However, in almost every resolver that interacts with a table using an integer-based primary key, I have to manually convert the ID string into an integer before performing any operations. This results in repetitive and cluttered code across hundreds of resolvers.

I'm wondering if there's a way to extend the argument builder in Pothos GraphQL to streamline this process. Specifically, I'd like to introduce a property like `valueAsNumber` for ID arguments. When set to true, it would automatically attempt to convert the ID to a number or throw an error if the string is not a valid number.

Here’s a current example of what I’m doing:
```
builder.mutationField("toggleIsCustomerVip", (t) => {
  return t.prismaField({
    type: "orders",
    nullable: false,
    args: {
      orderId: t.arg.id({ required: true }),
    },
    resolve: (query, root, args) => {
      const id = parseIntOrThrowIgnoreNullish(args.orderId, "Invalid orderId");
      ...
    },
  });
});
```

What I'd like to achieve instead:
```
builder.mutationField("toggleIsCustomerVip", (t) => {
  return t.prismaField({
    type: "orders",
    nullable: false,
    args: {
      orderId: t.arg.id({
        required: true,
        valueAsNumber: true,
      }),
    },
    resolve: (query, root, args) => {
      // args.orderId is now typeof number
      return toggleIsCustomerVip(args.orderId);
    },
  });
});
```

This would significantly reduce clutter and make the code more concise. Is there any way to accomplish this or something similar in Pothos?

**Comment by hayes** [comment, marked as answer]
You can build a plugin to add an intID method to the schema builder like this:

https://stackblitz.com/edit/typescript-dftsbi?file=index.ts

**Reply by solokhind** [reply]
I'll have to try that out. Thank you!
